src/services/domainService.ts

```diff
@@ -8,31 +8,55 @@ export abstract class DomainService<T> {
   }
   abstract listDomains(
     findOptions: Maybe<DomainFindOptions>
-  ): Promise<PaginationResponse<Domain>>;
+  ): Promise<PaginationResponse<DomainDto>>;
   abstract getDomain(
     id: string,
     findOptions: Maybe<DomainFindOptions>
-  ): Promise<Domain>;
+  ): Promise<DomainDto>;
   abstract getSubdomains(
     id: string,
     findOptions: Maybe<DomainFindOptions>
-  ): Promise<PaginationResponse<Domain>>;
+  ): Promise<PaginationResponse<DomainDto>>;
   abstract searchDomainsByOwner(
     address: string,
     findOptions: Maybe<DomainFindOptions>
-  ): Promise<PaginationResponse<Domain>>;
+  ): Promise<PaginationResponse<DomainDto>>;
   abstract searchDomainsByName(
     searchTerm: string,
     findOptions: Maybe<DomainFindOptions>
-  ): Promise<PaginationResponse<Domain>>;
-  abstract doServiceOperation(
-    serviceFn: Function
-  ): Promise<PaginationResponse<Domain> | Domain>;
+  ): Promise<PaginationResponse<DomainDto>>;
+  abstract doServiceOperation(serviceFn: Function): Promise<Domain[] | Domain>;
 
-  domainsToDomainsResponse(domains: Domain[]): PaginationResponse<Domain> {
-    const response: PaginationResponse<Domain> = {
+  domainsToPaginationResponse(
+    domains: ValidDomain[]
+  ): PaginationResponse<DomainDto> {
+    const domainResponses = domains.map((x) => this.domainToDomainDto(x));
+    const response: PaginationResponse<DomainDto> = {
       numResults: domains.length,
-      results: domains,
+      results: domainResponses,
+    };
+    return response;
+  }
+
+  domainToDomainDto(domain: ValidDomain): DomainDto {
+    const response: DomainDto = {
+      domainId: domain.domainId,
+      isRoot: domain.isRoot,
+      isValid: domain.isValid,
+      registrar: domain.registrar,
+      label: domain.label,
+      name: domain.name,
+      parent: domain.parent,
+      labelHash: domain.labelHash,
+      minter: domain.minter,
+      owner: domain.owner?.value,
```

Comment:
Now that Ryan has finished the blocking user story, do these still need to check null-ability with `?`

Reply:
Its still technically possible to have these fields as undefined, and since many of the functions are list views if one item is undefined for the whole page, it would cause the function to be unable to finish the request. 

So I would prefer to keep the nullable operator there for the nested properties

Reply:
If it's possible for them to be undefined then that should be marked in the `DomainDto` interface, otherwise having them as undefined here will just cause a crash/bug later in the chain because it's expected that the values are defined

Reply:
After the updates from my story. You don't need to check nullability on those fields.

Reply:
I cannot merge it in without these nullable fields, as the list view method will crash. It wont even work against the v2 deployment as there are many of these nested fields without proper values.

Perhaps I can just default it to something instead? That way the function will work and response can always expect a value

Reply:
Resolved in a call, this will work with the v2 deployment. We can remove the null checks.